tests/in/access.param.ron

```diff
@@ -5,7 +5,7 @@
 		adjust_coordinate_space: false,
 	),
 	msl: (
-		lang_version: (2, 0),
+		lang_version: (1, 2),
```

Comment:
Here and elsewhere: why does this require 1.2?

Reply:
- `access`
  We seem to be unable to use a `read_write` storage buffer in fragment stage prior to 1.2 ([see validation errors](https://github.com/gfx-rs/naga/actions/runs/6226374483/job/16898775262)).

  https://github.com/gfx-rs/naga/blob/54df410b949b36ef4fecc5a5e3949c77bb849b27/tests/in/access.wgsl#L145-L149

  This is not documented in the MSL spec but is mentioned [here](https://developer.apple.com/library/archive/documentation/Miscellaneous/Conceptual/MetalProgrammingGuide/WhatsNewiniOS10tvOS10andOSX1012/WhatsNewiniOS10tvOS10andOSX1012.html) (look for section "Function Buffer Read-Writes") that this became available in a feature sets that are also the first to support MSL 1.2.
- `bounds-check-image-restrict` & `bounds-check-image-rzsw`
  We seem to be unable to use `write` textures in fragment stage prior to 1.2 ([see validation errors](https://github.com/gfx-rs/naga/actions/runs/6226645516/job/16899569509)).
  This is also not documented in the MSL spec but is mentioned [here (same link as above)](https://developer.apple.com/library/archive/documentation/Miscellaneous/Conceptual/MetalProgrammingGuide/WhatsNewiniOS10tvOS10andOSX1012/WhatsNewiniOS10tvOS10andOSX1012.html) (look for section "Function Texture Read-Writes") that this became available in a feature set that is also the first to support MSL 1.2.
- `dualsource`
  We have validation for this and is also mentioned in the MSL spec.
  https://github.com/gfx-rs/naga/blob/54df410b949b36ef4fecc5a5e3949c77bb849b27/src/back/msl/mod.rs#L255

Reply:
Added the undocumented checks in c96fcd3e0fadb49e5e58ef243d5c073f7328fa3f.